kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js

```diff
@@ -663,6 +683,97 @@ define([
+         * 3. Send those to just the inputs those came from.
+         * 4. Up to widgets to act accordingly
+         * 5. Return the row ids that are duplicates
+         */
+        function updateDuplicateOutputValues() {
+            const outputValueCounts = model.getItem('outputParamIds').reduce((counts, paramId) => {
+                const outputValues = getParameterValuesByRow(paramId);
+                for (const [rowId, value] of Object.entries(outputValues)) {
+                    if (!value) {
+                        return counts;
+                    } else if (!(value in counts)) {
+                        counts[value] = [];
+                    }
+                    counts[value].push({ rowId, paramId });
+                }
+                return counts;
+            }, {});
+            const values = Object.keys(outputValueCounts).reduce((duplicateValues, value) => {
+                if (outputValueCounts[value].length > 1) {
+                    duplicateValues[value] = outputValueCounts[value];
+                }
+                return duplicateValues;
+            }, {});
+            // values should now look like:
+            // {
+            //   'field_value': [{ rowId1, paramId }, { rowId2, paramId }]
+            // }
+            // with only entries where there are at least two row ids
+            // translate to something easier to work with -
+            // {
+            //    rowId: {
+            //      paramId1: {
+            //        widget: Object
+            //        duplicateRows: [rowId, rowId]
+            //      },
+            //      paramId2: {
+            //        widget: Object,
+            //        duplicateRows: [rowId, rowId]
+            //    }
+            // }
+            const duplicates = {};
+            Object.values(values).forEach((dupArr) => {
+                const allRows = dupArr.reduce((rowIds, entry) => {
+                    rowIds.add(dataModel.rowIdToIndex[entry.rowId] + 1);
+                    return rowIds;
+                }, new Set());
+                dupArr.forEach((entry) => {
+                    const rowId = entry.rowId;
+                    const rowIdx = dataModel.rowIdToIndex[rowId] + 1;
+                    const paramInfo = {
+                        widget: dataModel.rows[rowId].widgets[entry.paramId],
+                        duplicateRows: [...allRows].filter((x) => x !== rowIdx),
+                    };
+                    if (paramInfo.duplicateRows.length === 0) {
+                        paramInfo.duplicateRows.push(rowIdx);
+                    }
+                    if (!(rowId in duplicates)) {
+                        duplicates[rowId] = {};
+                    }
+                    duplicates[rowId][entry.paramId] = paramInfo;
+                });
+            });
+            const outputParams = new Set(getAllOutputParameterWidgets());
+
+            Object.values(duplicates).forEach((duplicateEntry) => {
```

Comment:
Take advantage of `Sets` here. We're tracking the widget with a duplicate value in the `duplicates` object, so once we call `setDuplicateValue` on that particular widget, we can remove the widget from the set of all widgets that we just got. The remaining widgets in that set don't have any duplicates, so we call `clearDuplicateValue` on the rest. This'll clear up any widgets that were duplicate, but no longer are.